src/pandablocks_ioc/_hdf_ioc.py

```diff
@@ -202,10 +248,20 @@ async def _handle_hdf5_data(self) -> None:
             pipeline: List[Pipeline] = create_default_pipeline(
                 iter([self._get_filename()])
             )
-            flush_period: float = self._flush_period_record.get()
+
+            flush_period: Optional[float] = None
+            flush_event: Optional[asyncio.Event] = None
+            flush_mode = FlushMode(self._flush_mode_record.get())
+            if flush_mode == FlushMode.PERIODIC:
+                flush_period: float = self._flush_period_record.get()
```

Comment:
It's unusual to redefine the variable's type, and I don't think its necessary here